src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Lookup.cs

```diff
@@ -101,25 +128,54 @@ public override bool IsServerDelegatable(CallNode callNode, TexlBinding binding)
                 metadata = dataSource.DelegationMetadata.FilterDelegationMetadata;
             }
 
-            var args = callNode.Args.Children.VerifyValue();
-            if (args.Count > 2 && binding.IsDelegatable(args[2]))
-            {
-                SuggestDelegationHint(args[2], binding);
-                return false;
-            }
+            return true;
+        }
 
-            if (args.Count < 2)
+        public override bool IsEcsExcemptedLambda(int index)
+        {
+            // Only the second argument for lookup is an ECS excempted lambda
+            return index == 1;
+        }
+
+        public override ICallNodeDelegatableNodeValidationStrategy GetCallNodeDelegationStrategy()
+        {
+            return new LookUpCallNodeDelegationStrategy(this);
+        }
+
+        public bool IsValidDelegatableReductionNode(CallNode callNode, TexlNode reductionNode, TexlBinding binding)
+        {
+            if (!TryGetFilterOpDelegationMetadata(callNode, binding, out var metadata))
             {
                 return false;
             }
 
-            return IsValidDelegatableFilterPredicateNode(args[1], binding, metadata);
+            // use a variation of the filter predicate logic to determine if the reduction formula is delegatable, without enforcing the return type must be boolean
+            return IsValidDelegatableFilterPredicateNode(reductionNode, binding, metadata, generateHints: false, enforceBoolean: false);
         }
```

Comment:
Maybe I misunderstood, but I thought we had said that the reduction formula was going to always be evaluated locally?
Why would it share anything with the filter predicate logic?

Reply:
When we had our discussion, I had missed a critical element.  The primary issue is that the LookUp can be used with a reduction formula inside another Filter or LookUp call.  We properly delegate trivial reduction formulas in that scenario today, so we can't break that.  My proposed fix is to use a flavor of the same logic that determines whether a filter predicate is delegable to determine if the reduction can be code gen'd properly